pkg/policy/rule.go

```diff
@@ -74,26 +75,70 @@ func (r *rule) validate() error {
 	return nil
 }
 
-func mergeL4Port(ctx *SearchContext, r api.PortRule, p api.PortProtocol, dir string, proto string, resMap L4PolicyMap) int {
-	fmt := p.Port + "/" + proto
-	v, ok := resMap[fmt]
+func adjustL4PolicyIfNeeded(fromEndpoints []api.EndpointSelector, policy *L4Filter) bool {
+
+	if len(policy.FromEndpoints) == 0 && len(fromEndpoints) > 0 {
+		log.Debugf("skipping this L4 policy since it is already covered by an existing policy.")
+		return true
+	}
+
+	if len(policy.FromEndpoints) > 0 && len(fromEndpoints) == 0 {
```

Comment:
What's the difference between this line and line 80?

Reply:
Line 80 is saying that if the policy as resolved so far has an empty "FromEndpoints" list then it already covers the endpoints specified by the L4 filter that is currently being processed. Nil FromEndpoints would mean any endpoint can be used.

This line 85 is stating that if there are currently constraints on which endpoints this rule must receive traffic from in the policy, but the L4 Filter being processed allows contact from any Endpoint, then we need to loosen the current policy's restrictions on endpoints.